### PR TITLE
Adding an improved null handling for the super type

### PR DESCRIPTION
**Pull request by kaijMueller**
Hi,
I found that the super type "remove" (type parameter is null) is not consistent across the methods. Therefore I improved it, added the necessary tests, fixed a new bug with the super type import handling and added the "bin folder" to the gitignore.

BR,
Kai

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,4 @@ docs/reference/src/main/docbook/en-US/version_info.xml
 .forge_settings
 .factorypath
 atlassian-ide-plugin.xml
+bin/
```
Comment (gastaldi) [inline .gitignore]: Isn't this entry ignoring https://github.com/forge/roaster/tree/master/dist/src/main/resources/bin?

Comment (kaijMueller) [reply .gitignore]: Yes, next commit will fix this

File: .gitignore [hunk 2]
```diff
@@ -14,3 +14,4 @@ docs/reference/src/main/docbook/en-US/version_info.xml
 .forge_settings
 .factorypath
 atlassian-ide-plugin.xml
+/*/bin/
```
Comment (gastaldi) [inline .gitignore]: In fact you shouldn't have a bin folder if you're compiling using Maven. On what situations do you see a bin folder in the project root?

Comment (kaijMueller) [reply .gitignore]: It was auto-genercate by eclipse, I guess

**Pull request merged**